src/mrinufft/operators/base.py

```diff
@@ -423,11 +423,12 @@ def compute_density(self, method=None):
             method = get_density(method)
         if not callable(method):
             raise ValueError(f"Unknown density method: {method}")
-        self.density_method = lambda samples, shape: method(
-            samples,
-            shape,
-            **kwargs,
-        )
+        if self.density_method is None:
+            self.density_method = lambda samples, shape: method(
+                samples,
+                shape,
+                **kwargs,
+            )
```

Comment:
I think it would be best to make _density_method a private attribute, the user should not have to interact with it.